datafusion/optimizer/src/optimizer.rs

```diff
@@ -216,23 +216,24 @@ impl Optimizer {
             // subqueries to joins
             Arc::new(SimplifyExpressions::new()),
             Arc::new(RewriteDisjunctivePredicate::new()),
+            Arc::new(EliminateLimit::new()),
             Arc::new(EliminateFilter::new()),
+            // Filters can't be pushed down past Limits, we should do PushDownFilter after PushDownLimit
```

Comment:
are there dependencies between some optimization rules?

I don't why we use the same rule twice, such as `PushDownLimit`